temporal-sdk/src/main/java/io/temporal/internal/sync/ActivityInvocationHandler.java

```diff
@@ -54,8 +60,13 @@ private ActivityInvocationHandler(
   protected Function<Object[], Object> getActivityFunc(
       Method method, MethodRetry methodRetry, String activityName) {
     Function<Object[], Object> function;
+    ActivityOptions methodOptions =
+        (activityMethodOptions == null) ? null : activityMethodOptions.get(method.getName());
```

Comment:
I think we want to use the activity type name as the key to the map, not the method name.

Reply:
I remember you saying that keying from the actual Method object would be best and it's very conveniently here. Should we do that instead?

Reply:
It is not as we planned to support the configuration of these options through a config file. In this case activity type name makes much more sense than the method name.